diff --git a/.gitignore b/.gitignore
new file mode 100644
index 0000000..d978874
--- /dev/null
+++ b/.gitignore
@@ -0,0 +1,3 @@
+.swp
+.*.swp
+build
diff --git a/CMakeLists.txt b/CMakeLists.txt
index a0c1c35..55438a2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -15,7 +15,10 @@ cmake_minimum_required(VERSION 2.6)
 
 project (cfm)
 
-set (CFM_OUTPUT_DIR ${CMAKE_CURRENT_SOURCE_DIR}/bin CACHE STRING "Where do you want the output binaries to install to?" )
+if (NOT CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+	set (CFM_OUTPUT_DIR ${CMAKE_CURRENT_SOURCE_DIR}/bin CACHE STRING "Where do you want the output binaries to install to?" )
+	# For macOS, just use the standard CMAKE_INSTALL_PREFIX variable that you can set on command line. See <https://cmake.org/cmake/help/v3.0/command/install.html>.
+endif()
 
 #It is recommended, but not strictly necessary, to build and run the tests
 set( INCLUDE_TESTS OFF CACHE BOOL "Do you want to build the tests?" )
@@ -34,7 +37,11 @@ include_directories( ${Boost_INCLUDE_DIRS} )
 set (CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${CMAKE_CURRENT_SOURCE_DIR} )
 include (FindRDKit)
 if (RDKIT_FOUND)
-    include_directories( ${RDKIT_INCLUDE_DIR} ${RDKIT_INCLUDE_EXT_DIR} )
+	if (CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+    	include_directories( ${RDKIT_INCLUDE_DIR} )
+	else()
+    	include_directories( ${RDKIT_INCLUDE_DIR} ${RDKIT_INCLUDE_EXT_DIR} )
+	endif()
     link_directories ( ${RDKIT_LIBRARY_DIR} )
 else ()
     message (FATAL_ERROR "Could not find RDKit" )
diff --git a/FindRDKit.cmake b/FindRDKit.cmake
index 3e5d740..b3a564c 100644
--- a/FindRDKit.cmake
+++ b/FindRDKit.cmake
@@ -37,7 +37,21 @@ else()
         C:\\RDKit\\include
         C:\\RDKit\\External
       )
-    else()
+    elseif(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+      find_path(RDKIT_INCLUDE_DIR GraphMol/RDKitBase.h
+        PATHS
+          ${RDKIT_DIR}/Code
+          $ENV{RDKIT_INCLUDE_DIR}
+          $ENV{RDKIT_INCLUDE_PATH}
+          $ENV{RDKIT_BASE}/Code
+          $ENV{RDBASE}/Code
+          /usr/local/rdkit/include/Code
+          /usr/local/rdkit/include
+          /usr/local/include/rdkit
+          /usr/local/rdkit/Code
+          ~/rdkit/Code
+      )
+    else() # Other UNIX systems
       find_path(RDKIT_INCLUDE_DIR GraphMol/RDKitBase.h
         PATHS
           ${RDKIT_DIR}/Code
@@ -138,10 +152,20 @@ else()
     endif()
   endif()
 
-  if(RDKIT_INCLUDE_DIR AND RDKIT_INCLUDE_EXT_DIR AND RDKIT_LIBRARIES)
-    set(RDKIT_FOUND TRUE)
+  if (CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+  	  if(RDKIT_INCLUDE_DIR AND RDKIT_LIBRARIES)
+    	set(RDKIT_FOUND TRUE)
+  	endif()
+  else ()
+  	  if(RDKIT_INCLUDE_DIR AND RDKIT_INCLUDE_EXT_DIR AND RDKIT_LIBRARIES)
+    	set(RDKIT_FOUND TRUE)
+  	endif()
   endif()
 
   mark_as_advanced(RDINCHI_LIB INCHI_LIB GASTEIGER_LIB SUBSTRUCT_LIB RDGENERAL_LIB RDGEOMETRYLIB_LIB GRAPHMOL_LIB DEPICTOR_LIB SMILESPARSE_LIB FILEPARSERS_LIB)
-  mark_as_advanced(RDKIT_INCLUDE_DIR RDKIT_INCLUDE_EXT_DIR RDKIT_LIBRARIES)
+  if (CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+  	mark_as_advanced(RDKIT_INCLUDE_DIR RDKIT_LIBRARIES)
+  else ()
+  	mark_as_advanced(RDKIT_INCLUDE_DIR RDKIT_INCLUDE_EXT_DIR RDKIT_LIBRARIES)
+  endif()
 endif()
diff --git a/cfm-annotate/CMakeLists.txt b/cfm-annotate/CMakeLists.txt
index 08cb930..5943e07 100644
--- a/cfm-annotate/CMakeLists.txt
+++ b/cfm-annotate/CMakeLists.txt
@@ -12,11 +12,16 @@ set ( SRC_FILES  main.cpp )
 add_executable ( cfm-annotate ${SRC_FILES} )
 target_link_libraries ( cfm-annotate cfm-code ${REQUIRED_LIBS} )
 
-install ( TARGETS cfm-annotate 
-          DESTINATION ${CFM_OUTPUT_DIR} )
+if (CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+	install ( TARGETS cfm-annotate 
+          	  DESTINATION bin )
+else()
+	install ( TARGETS cfm-annotate 
+          	  DESTINATION ${CFM_OUTPUT_DIR} )
+endif()
           
 if (WIN32)
     install ( FILES ${LPSOLVE_INCLUDE_DIR}/lpsolve55.dll
               DESTINATION ${CFM_OUTPUT_DIR} )
 endif (WIN32)              
-          
\ No newline at end of file
+          
diff --git a/cfm-code/CMakeLists.txt b/cfm-code/CMakeLists.txt
index 29f46e1..9330916 100644
--- a/cfm-code/CMakeLists.txt
+++ b/cfm-code/CMakeLists.txt
@@ -52,6 +52,3 @@ endif ()
 
 add_library ( cfm-code STATIC ${SRC_FILES} ${HEADER_FILES} )
 target_link_libraries ( cfm-code ${REQUIRED_LIBS} )
-
-install ( FILES ${CMAKE_CURRENT_SOURCE_DIR}/ISOTOPE.DAT
-          DESTINATION ${CFM_OUTPUT_DIR} )    
\ No newline at end of file
diff --git a/cfm-code/FragmentGraphGenerator.cpp b/cfm-code/FragmentGraphGenerator.cpp
index e217d2f..cc24a9b 100644
--- a/cfm-code/FragmentGraphGenerator.cpp
+++ b/cfm-code/FragmentGraphGenerator.cpp
@@ -26,7 +26,11 @@
 #include <GraphMol/RingInfo.h>
 #include <GraphMol/SmilesParse/SmilesParse.h>
 #include <GraphMol/SmilesParse/SmilesWrite.h>
+#ifdef __APPLE__
+#include <GraphMol/inchi.h>
+#else
 #include <INCHI-API/inchi.h>
+#endif
 
 
 
diff --git a/cfm-code/ISOTOPE.DAT b/cfm-code/ISOTOPE.DAT
deleted file mode 100644
index 6870ca3..0000000
--- a/cfm-code/ISOTOPE.DAT
+++ /dev/null
@@ -1,517 +0,0 @@
-X  2
-1  0.9
-2  0.1
-
-H  2
-1.0078246  0.99985
-2.0141021  0.00015
-
-He  2
-3.01603    0.00000138
-4.00260    0.99999862
-
-Li  2
-6.015121   0.075
-7.016003   0.925
-
-Be  1
-9.012182   1.0
-
-B  2
-10.012937  0.199
-11.009305  0.801
-
-C  2
-12.0000000 0.988930
-13.0033554 0.011070
-
-N  2
-14.0030732 0.996337
-15.0001088 0.003663
-
-O  3
-15.9949141 0.997590
-16.9991322 0.000374
-17.9991616 0.002036
-
-F  1
-18.9984032 1.0
-
-Ne  3
-19.992435  0.9048
-20.993843  0.0027
-21.991383  0.0925
-
-Na  1
-22.989767  1.0
-
-Mg  3
-23.985042  0.7899
-24.985837  0.1000
-25.982593  0.1101
-
-Al  1
-26.981539  1.0
-
-Si  3
-27.976927  0.9223
-28.976495  0.0467
-29.973770  0.0310
-
-P  1
-30.973762  1.0
-
-S  4
-31.972070  0.9502
-32.971456  0.0075
-33.967866  0.0421
-35.967080  0.0002
-
-Cl  2
-34.9688531 0.755290
-36.9659034 0.244710
-
-Ar  3
-35.967545  0.00337
-37.962732  0.00063
-39.962384  0.99600
-
-K  3
-38.963707  0.932581
-39.963999  0.000117
-40.961825  0.067302
-
-Ca  6
-39.962591  0.96941
-41.958618  0.00647
-42.958766  0.00135
-43.955480  0.02086
-45.953689  0.00004
-47.952533  0.00187
-
-Sc  1
-44.955910  1.0
-
-Ti  5
-45.952629  0.080
-46.951764  0.073
-47.947947  0.738
-48.947871  0.055
-49.944792  0.054
-
-V  2
-49.947161  0.00250
-50.943962  0.99750
-
-Cr  4
-49.946046  0.04345
-51.940509  0.83790
-52.940651  0.09500
-53.938882  0.02365
-
-Mn  1
-54.938047  1.0
-
-Fe  4
-53.939612  0.0590
-55.934939  0.9172
-56.935396  0.0210
-57.933277  0.0028
-
-Co  1
-58.933198  1.0
-
-Ni  5
-57.935346  0.6827
-59.930788  0.2610
-60.931058  0.0113
-61.928346  0.0359
-63.927968  0.0091
-
-Cu  2
-62.939598  0.6917
-64.927793  0.3083
-
-Zn  5
-63.929145  0.486
-65.926034  0.279
-66.927129  0.041
-67.924846  0.188
-69.925325  0.006
-
-Ga  2
-68.925580  0.60108
-70.924700  0.39892
-
-Ge  5
-69.924250  0.205
-71.922079  0.274
-72.923463  0.078
-73.921177  0.365
-75.921401  0.078
-
-As  1
-74.921594  1.0
-
-Se  6
-73.922475  0.009
-75.919212  0.091
-76.919912  0.076
-77.9190    0.236
-79.916520  0.499
-81.916698  0.089
-
-Br  2
-78.918336  0.5069
-80.916289  0.4931
-
-Kr  6
-77.914     0.0035
-79.916380  0.0225
-81.913482  0.116
-82.914135  0.115
-83.911507  0.570
-85.910616  0.173
-
-Rb  2
-84.911794  0.7217
-86.909187  0.2783
-
-Sr  4
-83.913430  0.0056
-85.909267  0.0986
-86.908884  0.0700
-87.905619  0.8258
-
-Y  1
-88.905849  1.0
-
-Zr  5
-89.904703  0.5145
-90.905644  0.1122
-91.905039  0.1715
-93.906314  0.1738
-95.908275  0.0280
-
-Nb  1
-92.906377  1.0
-
-Mo  7
-91.906808  0.1484
-93.905085  0.0925
-94.905840  0.1592
-95.904678  0.1668
-96.906020  0.0955
-97.905406  0.2413
-99.907477  0.0963
-
-Tc  1
-98.0   1.0
-
-Ru  7
-95.907599  0.0554
-97.905287  0.0186
-98.905939  0.127
-99.904219  0.126
-100.905582  0.171
-101.904348  0.316
-103.905424  0.186
-
-Rh  1
-102.905500  1.0
-
-Pd  6
-101.905634  0.0102
-103.904029  0.1114
-104.905079  0.2233
-105.903478  0.2733
-107.903895  0.2646
-109.905167  0.1172
-
-Ag  2
-106.905092  0.51839
-108.904757  0.48161
-
-Cd  8
-105.906461  0.0125
-107.904176  0.0089
-109.903005  0.1249
-110.904182  0.1280
-111.902758  0.2413
-112.904400  0.1222
-113.903357  0.2873
-115.904754  0.0749
-
-In  2
-112.904061  0.043
-114.903880  0.957
-
-Sn  10
-111.904826  0.0097
-113.902784  0.0065
-114.903348  0.0036
-115.901747  0.1453
-116.902956  0.0768
-117.901609  0.2422
-118.903310  0.0858
-119.902200  0.3259
-121.903440  0.0463
-123.905274  0.0579
-
-Sb  2
-120.903821  0.574
-122.904216  0.426
-
-Te  8
-119.904048  0.00095
-121.903054  0.0259
-122.904271  0.00905
-123.902823  0.0479
-124.904433  0.0712
-125.903314  0.1893
-127.904463  0.3170
-129.906229  0.3387
-
-I  1
-126.904473  1.0
-
-Xe  9
-123.905894  0.0010
-125.904281  0.0009
-127.903531  0.0191
-128.904780  0.264
-129.903509  0.041
-130.905072  0.212
-131.904144  0.269
-133.905395  0.104
-135.907214  0.089
-
-Cs  1
-132.905429  1.0
-
-Ba  7
-129.906282  0.00106
-131.905042  0.00101
-133.904486  0.0242
-134.905665  0.06593
-135.904553  0.0785
-136.905812  0.1123
-137.905232  0.7170
-
-La  2
-137.90711   0.00090
-138.906347  0.99910
-
-Ce  4
-135.907140  0.0019
-137.905985  0.0025
-139.905433  0.8843
-141.909241  0.1113
-
-Pr  1
-140.907647  1.0
-
-Nd  7
-141.907719  0.2713
-142.909810  0.1218
-143.910083  0.2380
-144.912570  0.0830
-145.913113  0.1719
-147.916889  0.0576
-149.920887  0.0564
-
-Pm  1
-145.0  1.0
-
-Sm  7
-143.911998  0.031
-146.914895  0.150
-147.914820  0.113
-148.917181  0.138
-149.917273  0.074
-151.919729  0.267
-153.922206  0.227
-
-Eu  2
-150.919847  0.478
-152.921225  0.522
-
-Gd  7
-151.919786  0.0020
-153.920861  0.0218
-154.922618  0.1480
-155.922118  0.2047
-156.923956  0.1565
-157.924099  0.2484
-159.927049  0.2186
-
-Tb  1
-158.925342  1.0
-
-Dy  7
-155.925277  0.0006
-157.924403  0.0010
-159.925193  0.0234
-160.926930  0.189
-161.926795  0.255
-162.928728  0.249
-163.929171  0.282
-
-Ho  1
-164.930319  1.0
-
-Er  6
-161.928775  0.0014
-163.929198  0.0161
-165.930290  0.336
-166.932046  0.2295
-167.932368  0.268
-169.935461  0.149
-
-Tm  1
-168.934212  1.0
-
-Yb  7
-167.933894  0.0013
-169.934759  0.0305
-170.936323  0.143
-171.936378  0.219
-172.938208  0.1612
-173.938859  0.318
-175.942564  0.127
-
-Lu  2
-174.940770  0.9741
-175.942679  0.0259
-
-Hf  6
-173.940044  0.00162
-175.941406  0.05206
-176.943217  0.18606
-177.943696  0.27297
-178.945812  0.13629
-179.946545  0.35100
-
-Ta  2
-179.947462  0.00012
-180.947992  0.99988
-
-W  5
-179.946701  0.0012
-181.948202  0.263
-182.950220  0.1428
-183.950928  0.307
-185.954357  0.286
-
-Re  2
-184.952951  0.3740
-186.955744  0.6260
-
-Os  7
-183.952488  0.0002
-185.953830  0.0158
-186.955741  0.016
-187.955860  0.133
-188.958137  0.161
-189.958436  0.264
-191.961467  0.410
-
-Ir  2
-190.960584  0.373
-192.962917  0.627
-
-Pt  6
-189.959917  0.0001
-191.961019  0.0079
-193.962655  0.329
-194.964766  0.338
-195.964926  0.253
-197.967869  0.072
-
-Au  1
-196.966543  1.0
-
-Hg  7
-195.965807  0.0015
-197.966743  0.100
-198.968254  0.169
-199.968300  0.231
-200.970277  0.132
-201.970617  0.298
-203.973467  0.0685
-
-Tl  2
-202.972320  0.29524
-204.974401  0.70476
-
-Pb  4
-203.973020  0.014
-205.974440  0.241
-206.975872  0.221
-207.976627  0.524
-
-Bi  1
-208.980374  1.0
-
-Po  1
-209.0  1.0
-
-At  1
-210.0  1.0
-
-Rn  1
-222.0  1.0
-
-Fr  1
-223.0  1.0
-
-Ra  1
-226.025  1.0
-
-Ac  1
-227.028  1.0
-
-Th  1
-232.038054  1.0
-
-Pa  1
-231.0359  1.0
-
-U  3
-234.040946  0.000055
-235.043924  0.00720
-238.050784  0.992745
-
-Np  1
-237.048  1.0
-
-Pu  1
-244.0  1.0
-
-Am  1
-243.0  1.0
-
-Cm  1
-247.0  1.0
-
-Bk  1
-247.0  1.0
-
-Cf  1
-251.0  1.0
-
-Es  1
-252.0  1.0
-
-Fm  1
-257.0  1.0
-
-Md  1
-258.0  1.0
-
-No  1
-259.0  1.0
-
-Lr  1
-260.0  1.0
-
diff --git a/cfm-code/Isotope.cpp b/cfm-code/Isotope.cpp
index e5b5742..dc462b9 100644
--- a/cfm-code/Isotope.cpp
+++ b/cfm-code/Isotope.cpp
@@ -28,6 +28,525 @@ typedef unsigned long ulong;
 
 static const double DUMMY_MASS = -10000000;
 
+#define ISOTOPES \
+"X  2" "\n" \
+"1  0.9" "\n" \
+"2  0.1" "\n" \
+"" "\n" \
+"H  2" "\n" \
+"1.0078246  0.99985" "\n" \
+"2.0141021  0.00015" "\n" \
+"" "\n" \
+"He  2" "\n" \
+"3.01603    0.00000138" "\n" \
+"4.00260    0.99999862" "\n" \
+"" "\n" \
+"Li  2" "\n" \
+"6.015121   0.075" "\n" \
+"7.016003   0.925" "\n" \
+"" "\n" \
+"Be  1" "\n" \
+"9.012182   1.0" "\n" \
+"" "\n" \
+"B  2" "\n" \
+"10.012937  0.199" "\n" \
+"11.009305  0.801" "\n" \
+"" "\n" \
+"C  2" "\n" \
+"12.0000000 0.988930" "\n" \
+"13.0033554 0.011070" "\n" \
+"" "\n" \
+"N  2" "\n" \
+"14.0030732 0.996337" "\n" \
+"15.0001088 0.003663" "\n" \
+"" "\n" \
+"O  3" "\n" \
+"15.9949141 0.997590" "\n" \
+"16.9991322 0.000374" "\n" \
+"17.9991616 0.002036" "\n" \
+"" "\n" \
+"F  1" "\n" \
+"18.9984032 1.0" "\n" \
+"" "\n" \
+"Ne  3" "\n" \
+"19.992435  0.9048" "\n" \
+"20.993843  0.0027" "\n" \
+"21.991383  0.0925" "\n" \
+"" "\n" \
+"Na  1" "\n" \
+"22.989767  1.0" "\n" \
+"" "\n" \
+"Mg  3" "\n" \
+"23.985042  0.7899" "\n" \
+"24.985837  0.1000" "\n" \
+"25.982593  0.1101" "\n" \
+"" "\n" \
+"Al  1" "\n" \
+"26.981539  1.0" "\n" \
+"" "\n" \
+"Si  3" "\n" \
+"27.976927  0.9223" "\n" \
+"28.976495  0.0467" "\n" \
+"29.973770  0.0310" "\n" \
+"" "\n" \
+"P  1" "\n" \
+"30.973762  1.0" "\n" \
+"" "\n" \
+"S  4" "\n" \
+"31.972070  0.9502" "\n" \
+"32.971456  0.0075" "\n" \
+"33.967866  0.0421" "\n" \
+"35.967080  0.0002" "\n" \
+"" "\n" \
+"Cl  2" "\n" \
+"34.9688531 0.755290" "\n" \
+"36.9659034 0.244710" "\n" \
+"" "\n" \
+"Ar  3" "\n" \
+"35.967545  0.00337" "\n" \
+"37.962732  0.00063" "\n" \
+"39.962384  0.99600" "\n" \
+"" "\n" \
+"K  3" "\n" \
+"38.963707  0.932581" "\n" \
+"39.963999  0.000117" "\n" \
+"40.961825  0.067302" "\n" \
+"" "\n" \
+"Ca  6" "\n" \
+"39.962591  0.96941" "\n" \
+"41.958618  0.00647" "\n" \
+"42.958766  0.00135" "\n" \
+"43.955480  0.02086" "\n" \
+"45.953689  0.00004" "\n" \
+"47.952533  0.00187" "\n" \
+"" "\n" \
+"Sc  1" "\n" \
+"44.955910  1.0" "\n" \
+"" "\n" \
+"Ti  5" "\n" \
+"45.952629  0.080" "\n" \
+"46.951764  0.073" "\n" \
+"47.947947  0.738" "\n" \
+"48.947871  0.055" "\n" \
+"49.944792  0.054" "\n" \
+"" "\n" \
+"V  2" "\n" \
+"49.947161  0.00250" "\n" \
+"50.943962  0.99750" "\n" \
+"" "\n" \
+"Cr  4" "\n" \
+"49.946046  0.04345" "\n" \
+"51.940509  0.83790" "\n" \
+"52.940651  0.09500" "\n" \
+"53.938882  0.02365" "\n" \
+"" "\n" \
+"Mn  1" "\n" \
+"54.938047  1.0" "\n" \
+"" "\n" \
+"Fe  4" "\n" \
+"53.939612  0.0590" "\n" \
+"55.934939  0.9172" "\n" \
+"56.935396  0.0210" "\n" \
+"57.933277  0.0028" "\n" \
+"" "\n" \
+"Co  1" "\n" \
+"58.933198  1.0" "\n" \
+"" "\n" \
+"Ni  5" "\n" \
+"57.935346  0.6827" "\n" \
+"59.930788  0.2610" "\n" \
+"60.931058  0.0113" "\n" \
+"61.928346  0.0359" "\n" \
+"63.927968  0.0091" "\n" \
+"" "\n" \
+"Cu  2" "\n" \
+"62.939598  0.6917" "\n" \
+"64.927793  0.3083" "\n" \
+"" "\n" \
+"Zn  5" "\n" \
+"63.929145  0.486" "\n" \
+"65.926034  0.279" "\n" \
+"66.927129  0.041" "\n" \
+"67.924846  0.188" "\n" \
+"69.925325  0.006" "\n" \
+"" "\n" \
+"Ga  2" "\n" \
+"68.925580  0.60108" "\n" \
+"70.924700  0.39892" "\n" \
+"" "\n" \
+"Ge  5" "\n" \
+"69.924250  0.205" "\n" \
+"71.922079  0.274" "\n" \
+"72.923463  0.078" "\n" \
+"73.921177  0.365" "\n" \
+"75.921401  0.078" "\n" \
+"" "\n" \
+"As  1" "\n" \
+"74.921594  1.0" "\n" \
+"" "\n" \
+"Se  6" "\n" \
+"73.922475  0.009" "\n" \
+"75.919212  0.091" "\n" \
+"76.919912  0.076" "\n" \
+"77.9190    0.236" "\n" \
+"79.916520  0.499" "\n" \
+"81.916698  0.089" "\n" \
+"" "\n" \
+"Br  2" "\n" \
+"78.918336  0.5069" "\n" \
+"80.916289  0.4931" "\n" \
+"" "\n" \
+"Kr  6" "\n" \
+"77.914     0.0035" "\n" \
+"79.916380  0.0225" "\n" \
+"81.913482  0.116" "\n" \
+"82.914135  0.115" "\n" \
+"83.911507  0.570" "\n" \
+"85.910616  0.173" "\n" \
+"" "\n" \
+"Rb  2" "\n" \
+"84.911794  0.7217" "\n" \
+"86.909187  0.2783" "\n" \
+"" "\n" \
+"Sr  4" "\n" \
+"83.913430  0.0056" "\n" \
+"85.909267  0.0986" "\n" \
+"86.908884  0.0700" "\n" \
+"87.905619  0.8258" "\n" \
+"" "\n" \
+"Y  1" "\n" \
+"88.905849  1.0" "\n" \
+"" "\n" \
+"Zr  5" "\n" \
+"89.904703  0.5145" "\n" \
+"90.905644  0.1122" "\n" \
+"91.905039  0.1715" "\n" \
+"93.906314  0.1738" "\n" \
+"95.908275  0.0280" "\n" \
+"" "\n" \
+"Nb  1" "\n" \
+"92.906377  1.0" "\n" \
+"" "\n" \
+"Mo  7" "\n" \
+"91.906808  0.1484" "\n" \
+"93.905085  0.0925" "\n" \
+"94.905840  0.1592" "\n" \
+"95.904678  0.1668" "\n" \
+"96.906020  0.0955" "\n" \
+"97.905406  0.2413" "\n" \
+"99.907477  0.0963" "\n" \
+"" "\n" \
+"Tc  1" "\n" \
+"98.0   1.0" "\n" \
+"" "\n" \
+"Ru  7" "\n" \
+"95.907599  0.0554" "\n" \
+"97.905287  0.0186" "\n" \
+"98.905939  0.127" "\n" \
+"99.904219  0.126" "\n" \
+"100.905582  0.171" "\n" \
+"101.904348  0.316" "\n" \
+"103.905424  0.186" "\n" \
+"" "\n" \
+"Rh  1" "\n" \
+"102.905500  1.0" "\n" \
+"" "\n" \
+"Pd  6" "\n" \
+"101.905634  0.0102" "\n" \
+"103.904029  0.1114" "\n" \
+"104.905079  0.2233" "\n" \
+"105.903478  0.2733" "\n" \
+"107.903895  0.2646" "\n" \
+"109.905167  0.1172" "\n" \
+"" "\n" \
+"Ag  2" "\n" \
+"106.905092  0.51839" "\n" \
+"108.904757  0.48161" "\n" \
+"" "\n" \
+"Cd  8" "\n" \
+"105.906461  0.0125" "\n" \
+"107.904176  0.0089" "\n" \
+"109.903005  0.1249" "\n" \
+"110.904182  0.1280" "\n" \
+"111.902758  0.2413" "\n" \
+"112.904400  0.1222" "\n" \
+"113.903357  0.2873" "\n" \
+"115.904754  0.0749" "\n" \
+"" "\n" \
+"In  2" "\n" \
+"112.904061  0.043" "\n" \
+"114.903880  0.957" "\n" \
+"" "\n" \
+"Sn  10" "\n" \
+"111.904826  0.0097" "\n" \
+"113.902784  0.0065" "\n" \
+"114.903348  0.0036" "\n" \
+"115.901747  0.1453" "\n" \
+"116.902956  0.0768" "\n" \
+"117.901609  0.2422" "\n" \
+"118.903310  0.0858" "\n" \
+"119.902200  0.3259" "\n" \
+"121.903440  0.0463" "\n" \
+"123.905274  0.0579" "\n" \
+"" "\n" \
+"Sb  2" "\n" \
+"120.903821  0.574" "\n" \
+"122.904216  0.426" "\n" \
+"" "\n" \
+"Te  8" "\n" \
+"119.904048  0.00095" "\n" \
+"121.903054  0.0259" "\n" \
+"122.904271  0.00905" "\n" \
+"123.902823  0.0479" "\n" \
+"124.904433  0.0712" "\n" \
+"125.903314  0.1893" "\n" \
+"127.904463  0.3170" "\n" \
+"129.906229  0.3387" "\n" \
+"" "\n" \
+"I  1" "\n" \
+"126.904473  1.0" "\n" \
+"" "\n" \
+"Xe  9" "\n" \
+"123.905894  0.0010" "\n" \
+"125.904281  0.0009" "\n" \
+"127.903531  0.0191" "\n" \
+"128.904780  0.264" "\n" \
+"129.903509  0.041" "\n" \
+"130.905072  0.212" "\n" \
+"131.904144  0.269" "\n" \
+"133.905395  0.104" "\n" \
+"135.907214  0.089" "\n" \
+"" "\n" \
+"Cs  1" "\n" \
+"132.905429  1.0" "\n" \
+"" "\n" \
+"Ba  7" "\n" \
+"129.906282  0.00106" "\n" \
+"131.905042  0.00101" "\n" \
+"133.904486  0.0242" "\n" \
+"134.905665  0.06593" "\n" \
+"135.904553  0.0785" "\n" \
+"136.905812  0.1123" "\n" \
+"137.905232  0.7170" "\n" \
+"" "\n" \
+"La  2" "\n" \
+"137.90711   0.00090" "\n" \
+"138.906347  0.99910" "\n" \
+"" "\n" \
+"Ce  4" "\n" \
+"135.907140  0.0019" "\n" \
+"137.905985  0.0025" "\n" \
+"139.905433  0.8843" "\n" \
+"141.909241  0.1113" "\n" \
+"" "\n" \
+"Pr  1" "\n" \
+"140.907647  1.0" "\n" \
+"" "\n" \
+"Nd  7" "\n" \
+"141.907719  0.2713" "\n" \
+"142.909810  0.1218" "\n" \
+"143.910083  0.2380" "\n" \
+"144.912570  0.0830" "\n" \
+"145.913113  0.1719" "\n" \
+"147.916889  0.0576" "\n" \
+"149.920887  0.0564" "\n" \
+"" "\n" \
+"Pm  1" "\n" \
+"145.0  1.0" "\n" \
+"" "\n" \
+"Sm  7" "\n" \
+"143.911998  0.031" "\n" \
+"146.914895  0.150" "\n" \
+"147.914820  0.113" "\n" \
+"148.917181  0.138" "\n" \
+"149.917273  0.074" "\n" \
+"151.919729  0.267" "\n" \
+"153.922206  0.227" "\n" \
+"" "\n" \
+"Eu  2" "\n" \
+"150.919847  0.478" "\n" \
+"152.921225  0.522" "\n" \
+"" "\n" \
+"Gd  7" "\n" \
+"151.919786  0.0020" "\n" \
+"153.920861  0.0218" "\n" \
+"154.922618  0.1480" "\n" \
+"155.922118  0.2047" "\n" \
+"156.923956  0.1565" "\n" \
+"157.924099  0.2484" "\n" \
+"159.927049  0.2186" "\n" \
+"" "\n" \
+"Tb  1" "\n" \
+"158.925342  1.0" "\n" \
+"" "\n" \
+"Dy  7" "\n" \
+"155.925277  0.0006" "\n" \
+"157.924403  0.0010" "\n" \
+"159.925193  0.0234" "\n" \
+"160.926930  0.189" "\n" \
+"161.926795  0.255" "\n" \
+"162.928728  0.249" "\n" \
+"163.929171  0.282" "\n" \
+"" "\n" \
+"Ho  1" "\n" \
+"164.930319  1.0" "\n" \
+"" "\n" \
+"Er  6" "\n" \
+"161.928775  0.0014" "\n" \
+"163.929198  0.0161" "\n" \
+"165.930290  0.336" "\n" \
+"166.932046  0.2295" "\n" \
+"167.932368  0.268" "\n" \
+"169.935461  0.149" "\n" \
+"" "\n" \
+"Tm  1" "\n" \
+"168.934212  1.0" "\n" \
+"" "\n" \
+"Yb  7" "\n" \
+"167.933894  0.0013" "\n" \
+"169.934759  0.0305" "\n" \
+"170.936323  0.143" "\n" \
+"171.936378  0.219" "\n" \
+"172.938208  0.1612" "\n" \
+"173.938859  0.318" "\n" \
+"175.942564  0.127" "\n" \
+"" "\n" \
+"Lu  2" "\n" \
+"174.940770  0.9741" "\n" \
+"175.942679  0.0259" "\n" \
+"" "\n" \
+"Hf  6" "\n" \
+"173.940044  0.00162" "\n" \
+"175.941406  0.05206" "\n" \
+"176.943217  0.18606" "\n" \
+"177.943696  0.27297" "\n" \
+"178.945812  0.13629" "\n" \
+"179.946545  0.35100" "\n" \
+"" "\n" \
+"Ta  2" "\n" \
+"179.947462  0.00012" "\n" \
+"180.947992  0.99988" "\n" \
+"" "\n" \
+"W  5" "\n" \
+"179.946701  0.0012" "\n" \
+"181.948202  0.263" "\n" \
+"182.950220  0.1428" "\n" \
+"183.950928  0.307" "\n" \
+"185.954357  0.286" "\n" \
+"" "\n" \
+"Re  2" "\n" \
+"184.952951  0.3740" "\n" \
+"186.955744  0.6260" "\n" \
+"" "\n" \
+"Os  7" "\n" \
+"183.952488  0.0002" "\n" \
+"185.953830  0.0158" "\n" \
+"186.955741  0.016" "\n" \
+"187.955860  0.133" "\n" \
+"188.958137  0.161" "\n" \
+"189.958436  0.264" "\n" \
+"191.961467  0.410" "\n" \
+"" "\n" \
+"Ir  2" "\n" \
+"190.960584  0.373" "\n" \
+"192.962917  0.627" "\n" \
+"" "\n" \
+"Pt  6" "\n" \
+"189.959917  0.0001" "\n" \
+"191.961019  0.0079" "\n" \
+"193.962655  0.329" "\n" \
+"194.964766  0.338" "\n" \
+"195.964926  0.253" "\n" \
+"197.967869  0.072" "\n" \
+"" "\n" \
+"Au  1" "\n" \
+"196.966543  1.0" "\n" \
+"" "\n" \
+"Hg  7" "\n" \
+"195.965807  0.0015" "\n" \
+"197.966743  0.100" "\n" \
+"198.968254  0.169" "\n" \
+"199.968300  0.231" "\n" \
+"200.970277  0.132" "\n" \
+"201.970617  0.298" "\n" \
+"203.973467  0.0685" "\n" \
+"" "\n" \
+"Tl  2" "\n" \
+"202.972320  0.29524" "\n" \
+"204.974401  0.70476" "\n" \
+"" "\n" \
+"Pb  4" "\n" \
+"203.973020  0.014" "\n" \
+"205.974440  0.241" "\n" \
+"206.975872  0.221" "\n" \
+"207.976627  0.524" "\n" \
+"" "\n" \
+"Bi  1" "\n" \
+"208.980374  1.0" "\n" \
+"" "\n" \
+"Po  1" "\n" \
+"209.0  1.0" "\n" \
+"" "\n" \
+"At  1" "\n" \
+"210.0  1.0" "\n" \
+"" "\n" \
+"Rn  1" "\n" \
+"222.0  1.0" "\n" \
+"" "\n" \
+"Fr  1" "\n" \
+"223.0  1.0" "\n" \
+"" "\n" \
+"Ra  1" "\n" \
+"226.025  1.0" "\n" \
+"" "\n" \
+"Ac  1" "\n" \
+"227.028  1.0" "\n" \
+"" "\n" \
+"Th  1" "\n" \
+"232.038054  1.0" "\n" \
+"" "\n" \
+"Pa  1" "\n" \
+"231.0359  1.0" "\n" \
+"" "\n" \
+"U  3" "\n" \
+"234.040946  0.000055" "\n" \
+"235.043924  0.00720" "\n" \
+"238.050784  0.992745" "\n" \
+"" "\n" \
+"Np  1" "\n" \
+"237.048  1.0" "\n" \
+"" "\n" \
+"Pu  1" "\n" \
+"244.0  1.0" "\n" \
+"" "\n" \
+"Am  1" "\n" \
+"243.0  1.0" "\n" \
+"" "\n" \
+"Cm  1" "\n" \
+"247.0  1.0" "\n" \
+"" "\n" \
+"Bk  1" "\n" \
+"247.0  1.0" "\n" \
+"" "\n" \
+"Cf  1" "\n" \
+"251.0  1.0" "\n" \
+"" "\n" \
+"Es  1" "\n" \
+"252.0  1.0" "\n" \
+"" "\n" \
+"Fm  1" "\n" \
+"257.0  1.0" "\n" \
+"" "\n" \
+"Md  1" "\n" \
+"258.0  1.0" "\n" \
+"" "\n" \
+"No  1" "\n" \
+"259.0  1.0" "\n" \
+"" "\n" \
+"Lr  1" "\n" \
+"260.0  1.0" "\n" \
+"" "\n"
+
 void IsotopeCalculator::computeIsotopeSpectrum( Spectrum &output, const romol_ptr_t mol, long charge ){
 	
 	FormMap fm;
@@ -125,14 +644,9 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 */
 //-------------------------------------------------------------------------------
 
-void IsotopeCalculator::init_data( const char *filename ){
-  
-  std::ifstream f(filename);
+void IsotopeCalculator::init_data(){
   
-  if(f.fail()){
-      std::cout << "Error opening ISOTOPE.DAT file. Please ensure it is in the executable directory." << std::endl;
-	  throw IsotopeCalculationException();
-  }
+  std::istringstream f(ISOTOPES);
 
   sad.clear();
   em.clear();
@@ -173,7 +687,6 @@ void IsotopeCalculator::init_data( const char *filename ){
       break;
     }
   }  
-  f.close();
 }
 
 
@@ -317,4 +830,4 @@ void IsotopeCalculator::print_to_output(Spectrum & output, Pattern & result)
 
   if( output.size() == 0 ) throw EmptyIsotopeSpectrumException();
   
-}
\ No newline at end of file
+}
diff --git a/cfm-code/Isotope.h b/cfm-code/Isotope.h
index 8bb1d37..a61ec65 100644
--- a/cfm-code/Isotope.h
+++ b/cfm-code/Isotope.h
@@ -62,7 +62,7 @@ class IsotopeCalculationException: public std::exception{
 class IsotopeCalculator{
 
 public:
-	IsotopeCalculator( double a_intensity_thresh ) : verbose(0), intensity_thresh( a_intensity_thresh) { init_data("ISOTOPE.DAT"); };
+	IsotopeCalculator( double a_intensity_thresh ) : verbose(0), intensity_thresh( a_intensity_thresh) { init_data(); };
 	void computeIsotopeSpectrum( Spectrum &output, const romol_ptr_t mol, long charge );
 	void setVerbose(){ verbose = true; };
 	double getIntensityThresh() const { return intensity_thresh; };
@@ -76,7 +76,7 @@ private:
 	void setFormulaMap( FormMap &output, const romol_ptr_t mol );
 
 	//The remainder of the functions are copied directly from emass (with minor mods)
-	void init_data( const char *filename );
+	void init_data();
 	void convolute_basic(Pattern & h, const Pattern & g, const Pattern & f);
 	void prune(Pattern & f, double limit);
 	void calculate(Pattern & tmp, Pattern & result, FormMap & fm, double limit, long charge);
diff --git a/cfm-code/MolData.cpp b/cfm-code/MolData.cpp
index 72ca407..e5a78a0 100644
--- a/cfm-code/MolData.cpp
+++ b/cfm-code/MolData.cpp
@@ -28,7 +28,11 @@
 #include <GraphMol/RDKitBase.h>
 #include <GraphMol/Fingerprints/Fingerprints.h>
 #include <RDGeneral/types.h>
+#ifdef __APPLE__
+#include <GraphMol/inchi.h>
+#else
 #include <INCHI-API/inchi.h>
+#endif
 
 #include <string>
 #include <iostream>
diff --git a/cfm-code/Util.cpp b/cfm-code/Util.cpp
index 0e0bff6..3389872 100644
--- a/cfm-code/Util.cpp
+++ b/cfm-code/Util.cpp
@@ -21,7 +21,11 @@
 #include <GraphMol/MolOps.h>
 #include <GraphMol/PeriodicTable.h>
 #include <GraphMol/AtomIterators.h>
+#ifdef __APPLE__
+#include <GraphMol/inchi.h>
+#else
 #include <INCHI-API/inchi.h>
+#endif
 
 double getMassTol( double abs_tol, double ppm_tol, double mass ){
 	double mass_tol = (mass/1000000.0) * ppm_tol;
@@ -106,4 +110,4 @@ romol_ptr_t createMolPtr( const char* smiles_or_inchi ){
 	RDKit::ROMol *mol = static_cast<RDKit::ROMol *>(rwmol);
 	addIonicChargeLabels( mol ); 
 	return romol_ptr_t( mol );
-}
\ No newline at end of file
+}
diff --git a/cfm-id-precomputed/CMakeLists.txt b/cfm-id-precomputed/CMakeLists.txt
index 96b3926..72153a0 100644
--- a/cfm-id-precomputed/CMakeLists.txt
+++ b/cfm-id-precomputed/CMakeLists.txt
@@ -12,11 +12,16 @@ set ( SRC_FILES  main.cpp )
 add_executable ( cfm-id-precomputed ${SRC_FILES} )
 target_link_libraries ( cfm-id-precomputed cfm-code ${REQUIRED_LIBS} )
 
-install ( TARGETS cfm-id-precomputed 
-          DESTINATION ${CFM_OUTPUT_DIR} )
+if (CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+	install ( TARGETS cfm-id-precomputed 
+          	  DESTINATION bin )
+else()
+	install ( TARGETS cfm-id-precomputed 
+          	  DESTINATION ${CFM_OUTPUT_DIR} )
+endif()
           
 if (WIN32)
     install ( FILES ${LPSOLVE_INCLUDE_DIR}/lpsolve55.dll
               DESTINATION ${CFM_OUTPUT_DIR} )
 endif (WIN32)              
-          
\ No newline at end of file
+          
diff --git a/cfm-id/CMakeLists.txt b/cfm-id/CMakeLists.txt
index 0540836..0ff5ef5 100644
--- a/cfm-id/CMakeLists.txt
+++ b/cfm-id/CMakeLists.txt
@@ -12,11 +12,16 @@ set ( SRC_FILES  main.cpp )
 add_executable ( cfm-id ${SRC_FILES} )
 target_link_libraries ( cfm-id cfm-code ${REQUIRED_LIBS} )
 
-install ( TARGETS cfm-id 
-          DESTINATION ${CFM_OUTPUT_DIR} )
+if (CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+	install ( TARGETS cfm-id 
+          	  DESTINATION bin )
+else()
+	install ( TARGETS cfm-id 
+          	  DESTINATION ${CFM_OUTPUT_DIR} )
+endif()
           
 if (WIN32)
     install ( FILES ${LPSOLVE_INCLUDE_DIR}/lpsolve55.dll
               DESTINATION ${CFM_OUTPUT_DIR} )
 endif (WIN32)              
-          
\ No newline at end of file
+          
diff --git a/cfm-predict/CMakeLists.txt b/cfm-predict/CMakeLists.txt
index 52112d8..5008265 100644
--- a/cfm-predict/CMakeLists.txt
+++ b/cfm-predict/CMakeLists.txt
@@ -12,11 +12,16 @@ set ( SRC_FILES  main.cpp )
 add_executable ( cfm-predict ${SRC_FILES} )
 target_link_libraries ( cfm-predict cfm-code ${REQUIRED_LIBS} )
 
-install ( TARGETS cfm-predict
-          DESTINATION ${CFM_OUTPUT_DIR} )
+if (CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+	install ( TARGETS cfm-predict
+          	  DESTINATION bin )
+else()
+	install ( TARGETS cfm-predict
+          	  DESTINATION ${CFM_OUTPUT_DIR} )
+endif()
           
 if (WIN32)
     install ( FILES ${LPSOLVE_INCLUDE_DIR}/lpsolve55.dll
               DESTINATION ${CFM_OUTPUT_DIR} )
 endif (WIN32)              
-          
\ No newline at end of file
+          
diff --git a/cfm-test/CMakeLists.txt b/cfm-test/CMakeLists.txt
index 866706e..1eb02ab 100644
--- a/cfm-test/CMakeLists.txt
+++ b/cfm-test/CMakeLists.txt
@@ -24,11 +24,12 @@ set ( SRC_FILES  test.cpp
 add_executable ( cfm-test ${SRC_FILES} )
 target_link_libraries ( cfm-test cfm-code ${REQUIRED_LIBS} )
 
-install ( TARGETS cfm-test 
-          DESTINATION ${CFM_OUTPUT_DIR} )
-          
-install ( DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/tests/test_data
-          DESTINATION ${CFM_OUTPUT_DIR}/tests )          
-          
-          
-          
\ No newline at end of file
+if (NOT CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+	install ( TARGETS cfm-test 
+          	  DESTINATION ${CFM_OUTPUT_DIR} )
+          	  
+	install ( DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/tests/test_data
+          	  DESTINATION ${CFM_OUTPUT_DIR}/tests )          
+	# TODO For macOS, use cmake testing feature to run tests through `make test`. See <https://cmake.org/Wiki/CMake/Testing_With_CTest>. After, generalize for all operation systems.
+endif()
+
diff --git a/fraggraph-gen/CMakeLists.txt b/fraggraph-gen/CMakeLists.txt
index a24a244..3a8e4a5 100644
--- a/fraggraph-gen/CMakeLists.txt
+++ b/fraggraph-gen/CMakeLists.txt
@@ -12,11 +12,16 @@ set ( SRC_FILES  main.cpp )
 add_executable ( fraggraph-gen ${SRC_FILES} )
 target_link_libraries ( fraggraph-gen cfm-code ${REQUIRED_LIBS} )
 
-install ( TARGETS fraggraph-gen 
-          DESTINATION ${CFM_OUTPUT_DIR} )
+if (CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+	install ( TARGETS fraggraph-gen 
+          	  DESTINATION bin )
+else()
+	install ( TARGETS fraggraph-gen 
+          	  DESTINATION ${CFM_OUTPUT_DIR} )
+endif()
           
 if (WIN32)
     install ( FILES ${LPSOLVE_INCLUDE_DIR}/lpsolve55.dll
               DESTINATION ${CFM_OUTPUT_DIR} )
 endif (WIN32)              
-          
\ No newline at end of file
+          
